webapp/loggedin/modules/analysis/processingChain/actions.js

```diff
@@ -102,43 +115,72 @@ export const saveProcessingChain = () => async (dispatch, getState) => {
   const state = getState()
 
   const surveyId = SurveyState.getSurveyId(state)
-  const chain = R.pipe(ProcessingChainState.getProcessingChain, ProcessingChain.dissocTemporary)(state)
+  const surveyInfo = SurveyState.getSurveyInfo(state)
+  const surveyDefaultLang = Survey.getDefaultLanguage(surveyInfo)
+
   const step = R.pipe(
     ProcessingStepState.getProcessingStep,
     ProcessingStep.dissocTemporary,
     R.when(R.isEmpty, R.always(null)),
   )(state)
+
+  const stepValidation = step ? await ProcessingChainValidator.validateStep(step) : null
+
   const calculation = R.pipe(
     ProcessingStepCalculationState.getCalculation,
     ProcessingStepCalculation.dissocTemporary,
     R.when(R.isEmpty, R.always(null)),
   )(state)
+  const calculationValidation = await ProcessingChainValidator.validateCalculation(calculation, surveyDefaultLang)
```

Comment:
what is calculation is null? (you added the check to stepValidation)

Reply:
right, I'll check it